### PR TITLE
Add server TlsIdentityCtx helper

### PR DESCRIPTION
**Pull request by elmarco**
The snippet is generally useful to setup Tls/Hybrid server. Move it from example to server crate, with a "helper" feature gate.

### CODE REVIEW

File: crates/ironrdp/Cargo.toml [hunk 1]
```diff
@@ -47,18 +47,17 @@ ironrdp-dvc = { workspace = true, optional = true }
 ironrdp-rdpdr = { workspace = true, optional = true }
 ironrdp-rdpsnd = { workspace = true, optional = true }
 ironrdp-displaycontrol = { workspace = true, optional = true }
+x509-cert = "0.2.5"
```
Comment (CBenoit) [inline crates/ironrdp/Cargo.toml:50]: question: Do we really need this new dependency in the `ironrdp` crate? This used to be a dev dependency for the example, but now we have the helper from `ironrdp-server`.

File: crates/ironrdp/examples/server.rs [hunk 2]
```diff
@@ -389,10 +332,10 @@ async fn run(
 
     let server_builder = if let Some((cert_path, key_path)) = cert.as_deref().zip(key.as_deref()) {
         let identity = TlsIdentityCtx::init_from_paths(cert_path, key_path).context("failed to init TLS identity")?;
-        let acceptor = acceptor(identity.cert, identity.priv_key).context("failed to build TLS acceptor")?;
+        let acceptor = identity.make_acceptor().context("failed to build TLS acceptor")?;
 
         if hybrid {
-            server_builder.with_hybrid(acceptor, identity.pub_key)
+            server_builder.with_hybrid(acceptor, identity.pub_key().into())
```
Comment (CBenoit) [inline crates/ironrdp/examples/server.rs:338]: If the previous suggestion is applied, we can revert this to
```suggestion
            server_builder.with_hybrid(acceptor, identity.pub_key)
```

File: crates/ironrdp-server/src/helper.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+use std::{fs::File, io::BufReader, path::Path, sync::Arc};
+
+use anyhow::Context;
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use tokio_rustls::{rustls, TlsAcceptor};
+
+pub struct TlsIdentityCtx {
+    cert: rustls::pki_types::CertificateDer<'static>,
+    priv_key: rustls::pki_types::PrivateKeyDer<'static>,
+    pub_key: Vec<u8>,
```
[the remark this answers is not in the document]
Comment (elmarco) [reply crates/ironrdp-server/src/helper.rs]: ok

File: crates/ironrdp/Cargo.toml (hunk 1 above)
Comment (elmarco) [reply crates/ironrdp/Cargo.toml]: that's a mistake! thanks

**Review by CBenoit — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**